src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateClone = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateClone
+        = {
+            ...stateClone, ...action.extraData,
+          };
```

Comment:
in this approach - formatting:
```suggestion
        stateClone = {
          ...stateClone,
          ...action.extraData,
        };
```

but i'd suggest to not re-assign stateClone, but rather use 
`Object.assign(stateClone, action.extraData)`

and then you would declare it with `const`: `const stateClone = { ...state };`

Reply:
but yeah, for `clear` it's much easier to keep it as `let` to simply assign an empty object